src/sentry/snuba/sessions.py

```diff
@@ -573,3 +584,327 @@ def get_release_sessions_time_bounds(project_id, release, org_id, environments=N
+        * release_direction: Indicates which ordering should be used to
+            order releases' version by either ASC or DESC
+        * crash_free_function: optional arg that is passed when a function needs to be applied in query like in the
+            case of crash_free_sessions and crash_free_users
+        * environments
+    Return:
+        List of releases that either contains one release or none at all
+    """
+    release_conditions = [
+        Condition(Column("started"), Op.GTE, stats_start),
+        Condition(
+            Column("started"),
+            Op.LT,
+            datetime.utcnow(),
+        ),
+        Condition(Column("project_id"), Op.EQ, project_id),
+        Condition(Column("org_id"), Op.EQ, org_id),
+    ]
+    if environments is not None:
+        release_conditions.append(Condition(Column("environment"), Op.IN, environments))
+
+    # Get select statements and append to the select statement list a function if
+    # crash_free_option whether it is crash_free_users or crash_free_sessions is picked
+    select = [
+        Column("project_id"),
+        Column("release"),
+    ]
+    if crash_free_function is not None:
+        select.append(crash_free_function)
+
+    having = [
+        Or(
+            conditions=[
+                Condition(Column(scope), scope_operation, scope_value),
+                And(
+                    conditions=[
+                        Condition(Column(scope), Op.EQ, scope_value),
+                        Condition(Column("release"), release_operation, release),
+                    ]
+                ),
+            ]
+        )
+    ]
+    orderby = [
+        OrderBy(direction=scope_direction, exp=Column(scope)),
+        OrderBy(direction=release_direction, exp=Column("release")),
+    ]
+
+    query = (
+        Query(
+            dataset=Dataset.Sessions.value,
+            match=Entity("sessions"),
+            select=select,
+            where=release_conditions,
+            having=having,
+            groupby=[Column("release"), Column("project_id")],
+            orderby=orderby,
+        )
+        .set_limit(limit)
+        .set_offset(0)
+    )
+    return raw_snql_query(query, referrer="sessions.get_prev_or_next_release")["data"]
+
+
+def __get_prev_operation_and_direction(scope):
+    """
+    Helper function that based on the scope returns the appropriate operation and direction
+    required for getting the previous element or release according to that scope
+    Inputs:
+        * scope: Sort order criteria -> sessions, users, crash_free_sessions, crash_free_users
+    Returns:
+        A dictionary of keys scope_operation, scope_direction, release_operation and release_direction
+        that correspond to which operations and directions should be carried out in this particular scope's query and
+        the direction they should be ordered in
+    """
+    op_and_direction_dict = {
```

Comment:
We could probably just declare this at the module level and use it directly, rather than constructing it every time. Also wondering if there's a way to consolidate the next/prev dicts, it seems like they are just mirrors of each other, maybe there's something we can do there.